backend/akr/src/test/java/fi/oph/akr/scheduled/ExpiringAuthorisationsEmailCreatorTest.java

```diff
@@ -85,17 +86,38 @@ public void testCheckExpiringAuthorisations() {
     createAuthorisationTermReminder(remindedAuth2);
     createAuthorisationTermReminder(remindedAuth2);
 
-    emailCreator.checkExpiringAuthorisations();
+    final Translator t9 = createTranslator("t9@invalid");
+    final Authorisation qExpiringButHasEquivalentAuthorisationNotExpiring = createAuthorisation(t9, meetingDate, date);
+    final Authorisation qNotExpiringWhichPreventsEquivalentAuthorisationReminder = createAuthorisation(
+      t9,
+      meetingDate,
+      date.plusMonths(3).plusDays(1)
+    );
 
-    verify(clerkEmailService, times(3)).createAuthorisationExpiryEmail(longCaptor.capture());
+    final Translator t10 = createTranslator("t10@invalid");
+    final Authorisation qExpiringButHasEquivalentAuthorisationAlsoExpiring = createAuthorisation(
+      t10,
+      meetingDate,
+      date
+    );
+    final Authorisation qExpiringWhichPreventsEquivalentAuthorisationReminder = createAuthorisation(
+      t10,
+      meetingDate,
+      date.plusMonths(1)
+    );
 
-    final List<Long> expiringAuthIds = longCaptor.getAllValues();
+    emailCreator.checkExpiringAuthorisations();
 
-    assertEquals(3, expiringAuthIds.size());
+    verify(clerkEmailService, times(4)).createAuthorisationExpiryEmail(longCaptor.capture());
 
-    assertTrue(expiringAuthIds.contains(auth1.getId()));
-    assertTrue(expiringAuthIds.contains(auth2.getId()));
-    assertTrue(expiringAuthIds.contains(auth3.getId()));
+    final Set<Long> expiringAuthIds = Set.copyOf(longCaptor.getAllValues());
+    assertEquals(
+      Stream
+        .of(auth1, auth2, auth3, qExpiringWhichPreventsEquivalentAuthorisationReminder)
+        .map(Authorisation::getId)
+        .collect(Collectors.toSet()),
+      expiringAuthIds
```

Comment:
Meniköhän kikkailuksi? Aiemmin siis tarkistettiin ensin että expiringAuthsIds pituus on jotakin, sitten tarkistettiin mitä se sisältää. Vaarana on että testiä muutettaessa jää tarkistamatta onko jokin id listassa vai ei.

Reply:
Failaahan tämä assert jatkossakin, jos testi muuttuu ja sitä kautta lista auktorisointien id:eitä, joilla tuota `createAuthorisationExpiryEmail` metodia kutsutaan? Musta ok näin ellei nyt jää jotain tajuamatta.